mlir/test/Conversion/VectorToSCF/vector-to-scf.mlir

```diff
@@ -748,3 +749,16 @@ func.func @cannot_fully_unroll_transfer_write_of_nd_scalable_vector(%vec: vector
   vector.transfer_write %vec, %memref[%c0, %c0] {in_bounds = [true, true]} : vector<[4]x[4]xf32>, memref<?x?xf32>
   return
 }
+
+//  -----
+
+// TARGET-RANK-ZERO-LABEL: func @unroll_transfer_write_target_rank_zero
+func.func @unroll_transfer_write_target_rank_zero(%vec : vector<2xi32>) {
+  // TARGET-RANK-ZERO: %[[EXTRACTED:.*]] = vector.extract
+  // TARGET-RANK-ZERO: %[[BROADCASTED:.*]] = vector.broadcast %[[EXTRACTED]]
+  // TARGET-RANK-ZERO: vector.transfer_write %[[BROADCASTED]]
```

Comment:
nit: I would find it helpful if the types and both extract/broadcast/transfer_write sequences were CHECK'ed